src/dbus.c

```diff
@@ -604,6 +604,9 @@ static struct notification *dbus_message_to_notification(const gchar *sender, GV
                 g_variant_unref(dict_value);
         }
 
+        // Set the dbus timeout
+        n->dbus_timeout = timeout < 0 ? timeout : ((gint64)timeout) * 1000;
```

Comment:
Instead of storing a separate dbus_timeout variable, I'd just override the timeout.

Reply:
```suggestion
if (timeout >= 0)
        n->dbus_timeout =((gint64)timeout) * 1000;
```

If you want to use the dbus_timeout in the future, I'd still suggestion doing this.
Also, I think it's better to modify n->timeout in notification.c then.